File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/repository/CategoryRepository.java
package com.iyoa.cleanaddis.repository;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

import com.iyoa.cleanaddis.model.Category;

public interface CategoryRepository extends JpaRepository<Category, UUID> {

	Category findCategoryByName(String name);

}

File: app/src/main/java/com/iyoa/cleanaddis/repository/common/CategoryRepository.kt
import androidx.lifecycle.LiveData
import com.iyoa.cleanaddis.data.common.Category
import com.iyoa.cleanaddis.data.news.CategoryDAO

class CategoryRepository(private val categoryDAO: CategoryDAO) {
    fun allCategories(): LiveData<List<Category>> = categoryDAO.getAllCategory()

    fun insertMedia(category: Category){
        categoryDAO.insertCategory(category)
    }
    fun getCategoryByName(name: String): Category{
        return categoryDAO.getCategoryByName(name)
    }
}